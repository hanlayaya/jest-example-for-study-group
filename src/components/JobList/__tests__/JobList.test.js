/* eslint-disable import/first */
jest.mock('../JobContainer', () => {
  return 'Job';
});

import React from 'react';
import { mount } from 'enzyme';

describe('<JobList />', function () {
  let JobList;

  beforeEach(() => {
    jest.resetModules();
    JobList = require('../JobList').default;

    this.params = {
      loadIfNeeded: jest.fn(),
    };

    this.makeSubject = () => {
      const { jobIds, loadIfNeeded } = this.params;

      return mount(
        <JobList
          jobIds={jobIds}
          loadIfNeeded={loadIfNeeded}
        />
      );
    };
  });

  describe('when it has no any data', () => {
    beforeEach(() => {
      this.params.jobIds = [];

      this.subject = this.makeSubject();
    });

    it('should be same as snapshot', () => {
      expect(this.subject.html()).toMatchSnapshot();
    });
  });

  describe('when it has some jobs data', () => {
    beforeEach(() => {
      this.params.jobIds = [1, 2, 3];

      this.subject = this.makeSubject();
    });

    it('should be same as snapshot', () => {
      expect(this.subject.html()).toMatchSnapshot();
    });

    it('should has 3 items', () => {
      expect(this.subject.find('Job').length).toEqual(3);
    });
  });
});
