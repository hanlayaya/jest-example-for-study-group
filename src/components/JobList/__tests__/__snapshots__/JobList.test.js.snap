// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobList /> when it has no any data should be same as snapshot 1`] = `"<div>No data</div>"`;

exports[`<JobList /> when it has some jobs data should be same as snapshot 1`] = `"<div><job id=\\"1\\"></job><job id=\\"2\\"></job><job id=\\"3\\"></job></div>"`;
